File: Interview/ToDo/ToDo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ToDo
{
    internal class ToDoDTO
    {
        public string Name { get; set; }
        public bool Completed { get; set; }
        public int EstimatedHours { get; set; }
        public int CategoryId { get; set; }
    }

    internal class ToDo
    {
        //return a list of the dtos that are not completed
        public List<ToDoDTO> GetTodos()
        {
            return null;
        }

        //return a list of the names of the dtos that are not completed
        public List<string> GetTodoNames()
        {
            return null;
        }

        //return the dto with the shortest Estimated Hours
        public ToDoDTO GetEasiest()
        {
            return null;
        }

        private List<ToDoDTO> _todos = new() {
            new (){
                Name= "Clean Windows",
                Completed = true,
                EstimatedHours = 5,
                CategoryId = 1
            },
            new (){
                Name= "Vacume",
                Completed= false,
                EstimatedHours= 3,
                CategoryId= 1
            },
            new (){
                Name= "Dust",
                Completed= false,
                EstimatedHours= 1,
                CategoryId= 1
            },
            new (){
                Name= "Grocery Shopping",
                Completed= false,
                EstimatedHours= 1,
                CategoryId= 2
            }
        };
    }
}

File: Interview-TS/todo.ts
interface ToDo_DTO {
  name: string;
  completed: boolean;
  estimatedHours: number;
  categoryId: number;
}

class ToDo {
  //return a list of the dtos that are not completed
  public GetTodos(): ToDo_DTO[] {
    return null;
  }

  //return a list of the names of the dtos that are not completed
  public GetTodoNames(): string[] {
    return null;
  }

  //return the dto with the shortest Estimated Hours
  public GetEasiest(): ToDo_DTO {
    return null;
  }

  private _todos: ToDo_DTO[] = [
    {
      name: "Clean Windows",
      completed: true,
      estimatedHours: 5,
      categoryId: 1,
    },
    {
      name: "Vacume",
      completed: false,
      estimatedHours: 3,
      categoryId: 1,
    },
    {
      name: "Dust",
      completed: false,
      estimatedHours: 1,
      categoryId: 1,
    },
    {
      name: "Grocery Shopping",
      completed: false,
      estimatedHours: 1,
      categoryId: 2,
    },
  ];
}
